docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc

```diff
@@ -81,6 +81,12 @@ When you run either command, the following XML is added to your build file:
 implementation("io.quarkus:quarkus-security-jpa")
 ----
 
+=== Using Hibernate Reactive
```

Comment:
Hey @michalvavrik IMHO it would make sense to also show how to configure the application in this tutorial, below. 
Moreover, I think we should `deprecate` the non-reactive version. I.e refactor this doc to show the reactive version, and have the note for the users who may prefer to continue using the legacy extension

Reply:
> Hey @michalvavrik IMHO it would make sense to also show how to configure the application in this tutorial, below. 

@sberyozkin I'll look into it. There is not many differences to `security-jpa` configuration wise, you just need to change datasource url, activate reactive datasource and if using panache, change panache imports to reactive ones

> Moreover, I think we should `deprecate` the non-reactive version. I.e refactor this doc to show the reactive version, and have the note for the users who may prefer to continue using the legacy extension

No, that's not a case. `security-jpa` should be used by applications that are using `hibernate-orm` and `security-jpa-reactive` should be used by applications using `hibernate-reactive`. Hibernate reactive is not replacement for the Hibernate ORM. Basically, blocking part happens on drivers side, there are never going to be fully non-blocking (reactive) extensions supporting Hibernate ORM.

Reply:
@michalvavrik Well, there is obviously some advantage to using `hibernate-reactive`, from the docs:
```
Hibernate Reactive is a reactive API for Hibernate ORM, supporting *non-blocking database drivers* and a reactive style of interaction with the database.
```
it talks about the non-blocking drivers as well. In any case, my understanding that we want to encourage Quarkus users to use `hibernate-reactive` as opposed to the non-reactive hibernate API, so it makes sense IMHO to tune the current guide to refocus users on the `hibernate-reactive`/`security-jpa-reactive`. I'm not saying we should hide the fact  some users may need to use `hibernate-orm`/`security-jpa`, but I'm suggesting we should guide users toward the reactive path while also mentioning the `hibernate-orm`/`security-jpa` option (sure, lets not qualify it as `legacy` if you think it is not correct), as opposed to highlighting `hibernate-orm`/`security-jpa` as the main option with a short note about the more optimal  `hibernate-reactive`/`security-jpa-reactive` combo.

Reply:
@michalvavrik That said, may be we can handle it later, with the help from the Doc team, for now, please only provide an example how to configure with the reactive version, which should be enough

Reply:
> @michalvavrik That said, may be we can handle it later, with the help from the Doc team, for now, please only provide an example how to configure with the reactive version, which should be enough

sure thing @sberyozkin ; I believe I understand advantages of using hibernate reactive in this case (non-blocking endpoints don't need to switch threads etc.), however I'm not convinced there is general agreement on preferring Hibernate Reactive (f.e. in 1st Quarkus 3 final there will be just Hibernate Reactive beta use), there is way higher attention paid to Hibernate ORM atm. I agree with your proposal, let's handle it later with help from docs team. Thank you

Reply:
and IIRC this Hibernate Reactive beta only guarantees PostgreSQL to work, I created some issues and got feedback that rest with reliably work with the final.